src/database/models/event_item.rs

```diff
@@ -0,0 +1,285 @@
+                    creator_id: triggerer_id.map_or_else(|| {
+                        Err(DatabaseError::UnexpectedNull(
+                            "Neither triggerer_id nor triggerer_id_type should be null for project creation".to_string(),
+                        ))
+                    }, |v| v.try_into())?,
+                },
+                time: value.created.map_or_else(
+                    || {
+                        Err(DatabaseError::UnexpectedNull(
+                            "the value of created should not be null".to_string(),
+                        ))
+                    },
+                    |c| Ok(c),
+                )?,
+            },
+        })
+    }
+}
+
+impl Event {
+    pub async fn insert(
+        self,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    ) -> Result<(), DatabaseError> {
+        Self::insert_many(vec![self], transaction).await
+    }
+
+    pub async fn insert_many(
+        events: Vec<Self>,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    ) -> Result<(), DatabaseError> {
+        RawEvent::insert_many(
+            events.into_iter().map(|e| e.into()).collect_vec(),
+            transaction,
+        )
+        .await
+    }
+
+    pub async fn get_events(
+        target_selectors: &[EventSelector],
+        triggerer_selectors: &[EventSelector],
+        exec: impl sqlx::Executor<'_, Database = sqlx::Postgres>,
+    ) -> Result<Vec<Event>, DatabaseError> {
+        let (target_ids, target_id_types, target_event_types) =
+            unzip_event_selectors(target_selectors);
+        let (triggerer_ids, triggerer_id_types, triggerer_event_types) =
+            unzip_event_selectors(triggerer_selectors);
+        Ok(sqlx::query_as!(
+            RawEvent,
+            r#"
+            SELECT 
+                id,
+                target_id,
+                target_id_type as "target_id_type: _",
+                triggerer_id,
+                triggerer_id_type as "triggerer_id_type: _",
+                event_type as "event_type: _",
+                metadata,
+                created
+            FROM events e
+            WHERE 
+                (target_id, target_id_type, event_type) 
+                = ANY(SELECT * FROM UNNEST ($1::bigint[], $2::text[], $3::text[]))
+            OR
+                (triggerer_id, triggerer_id_type, event_type) 
+                = ANY(SELECT * FROM UNNEST ($4::bigint[], $5::text[], $6::text[]))
+            ORDER BY created DESC
+            "#,
+            &target_ids[..],
+            &target_id_types[..] as &[IdType],
+            &target_event_types[..] as &[EventType],
+            &triggerer_ids[..],
+            &triggerer_id_types[..] as &[IdType],
+            &triggerer_event_types[..] as &[EventType]
+        )
+        .fetch_all(exec)
+        .await?
+        .into_iter()
+        .map(|r| r.try_into())
+        .collect::<Result<Vec<_>, _>>()?)
```

Comment:
see clippy warning

Reply:
Fixed this and others, the two remaining clippy warnings are in code that's going to change.